packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts

```diff
@@ -0,0 +1,164 @@
+import cdk = require('@aws-cdk/core');
+import { Chain } from '../chain';
+import { StateGraph } from '../state-graph';
+import { CatchProps, IChainable, INextable, RetryProps } from '../types';
+import { StateType } from './private/state-type';
+import { renderJsonPath, State } from './state';
+
+/**
+ * Properties for defining a Map state
+ */
+export interface MapProps {
+    /**
+     * An optional description for this state
+     *
+     * @default No comment
+     */
+    readonly comment?: string;
+
+    /**
+     * JSONPath expression to select part of the state to be the input to this state.
+     *
+     * May also be the special value DISCARD, which will cause the effective
+     * input to be the empty object {}.
+     *
+     * @default $
+     */
+    readonly inputPath?: string;
+
+    /**
+     * JSONPath expression to select part of the state to be the output to this state.
+     *
+     * May also be the special value DISCARD, which will cause the effective
+     * output to be the empty object {}.
+     *
+     * @default $
+     */
+    readonly outputPath?: string;
+
+    /**
+     * JSONPath expression to indicate where to inject the state's output
+     *
+     * May also be the special value DISCARD, which will cause the state's
+     * input to become its output.
+     *
+     * @default $
+     */
+    readonly resultPath?: string;
+
+    /**
+     * JSONPath expression to select the array to iterate over
+     *
+     * @default $
+     */
+    readonly itemsPath?: string;
+
+    /**
+     * The JSON that you want to override your default iteration input
+     *
+     * @default $
+     */
+    readonly parameters?: { [key: string]: any };
```

Comment:
Don't see this parameter documented here: https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-map-state.html

Are you sure it exists?

Reply:
It does! You can use the [context object](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html#contextobject-map) in map state. I'll talk to our technical writer to see if we can clarify the docs here.